shuup/core/models/_shipments.py

```diff
@@ -58,11 +59,8 @@ def all_except_deleted(self):
         return self.exclude(status=ShipmentStatus.DELETED)
 
 
-class Shipment(models.Model):
-    order = models.ForeignKey(
-        "Order", blank=True, null=True, related_name='shipments', on_delete=models.PROTECT, verbose_name=_("order"))
-    supplier = models.ForeignKey(
-        "Supplier", related_name='shipments', on_delete=models.PROTECT, verbose_name=_("supplier"))
+# TODO (2.0): Rename this to Shipment
+class AbstractShipment(ShuupModel):
```

Comment:
Hmm... Is this AbstractShipment really needed?

I know, I know... I was there when we planned this, but I didn't notice then that the `order` field is actually *nullable*. So maybe the addon, which wants to create shipments which are not linked to order, can do so by setting `order=None` and linking the shipment to its own models with external one-to-one field (or FK/M2M).

Isn't it the supplier which updates the stocks? So probably every shipment should have a supplier. Or what do you think?

I'm worried that if an addon creates some FoobarShipments that are derived from AbstractShipment, then nothing will process them and product stock counts won't ever be updated.